api/client/connection.go

```diff
@@ -83,21 +97,30 @@ type Response struct {
 }
 
 func (c *Connection) Do(req *Request) (*Response, error) {
-	httpreq, err := req.HTTPRequest()
-	if err != nil {
-		return nil, err
-	}
-	res, err := c.client.Do(httpreq)
-	if err != nil {
-		return nil, err
-	}
-	if res.StatusCode < 200 || res.StatusCode > 299 {
-		err = parseError(res)
+	for i := 0; ; i++ {
+		httpreq, err := req.HTTPRequest()
+		if err != nil {
+			return nil, err
+		}
+		res, err := c.client.Do(httpreq)
+		if err != nil {
+			return nil, err
+		}
+		if res.StatusCode < 200 || res.StatusCode > 299 {
+			err = parseError(res)
+			if i == 0 && res.StatusCode == 401 && err.(*ErrorResponse).Err.Error() == "invalid token" {
+				var access string
+				if access, err = c.refreshAuthToken(req.ctx); err == nil {
+					req.Header.Set("Authorization", "Bearer "+access)
+					continue
+				}
```

Comment:
If `c.refreshAuthToken` fails, this will return a `Response` that doesn't match `err`.
```suggestion
				access, err = c.refreshAuthToken(req.ctx)
				if err != nil {
					return nil, err
				}
				req.Header.Set("Authorization", "Bearer "+access)
				continue
```